app/src/main/java/com/example/android/sunshine/app/SettingsActivity.java

```diff
@@ -147,4 +160,34 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
     public Intent getParentActivityIntent() {
         return super.getParentActivityIntent().addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
     }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == PLACE_PICKER_REQUEST) {
+            if (resultCode != RESULT_OK) {
+                return;
+            }
+            Place place = PlacePicker.getPlace(this, data);
+            String address = place.getAddress().toString();
+            LatLng latLong = place.getLatLng();
+            if (TextUtils.isEmpty(address)) {
+                address = String.format(Locale.getDefault(), "(%.2f, %.2f)", latLong.latitude, latLong.longitude);
+            }
+
+            SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
+            SharedPreferences.Editor editor = sharedPreferences.edit();
+            editor.putString(getString(R.string.pref_location_key), address);
+            editor.putFloat(getString(R.string.pref_location_latitude), (float) latLong.latitude);
+            editor.putFloat(getString(R.string.pref_location_longitude), (float) latLong.longitude);
+            editor.commit();
+
+            // Update UI and trigger sync adapter
+            Preference locationPreference = findPreference(getString(R.string.pref_location_key));
+            setPreferenceSummary(locationPreference, address);
+            PrefUtility.resetLocationStatus(this);
+            SunshineSyncAdapter.syncImmediately(this);
```

Comment:
all the above code could be a new method to keep the `onActivityResult` small since checking for an activity result and actually processing it are 2 separate concerns